mongoengine/connection.py

```diff
@@ -28,6 +30,16 @@ class MongoEngineConnectionError(Exception):
 _dbs = {}
 
 
+def check_db_name(name):
+    """Check if a database name is valid.
+    This functionality is copied from pymongo Database class constructor.
+    """
+    if not isinstance(name, string_type):
+        raise TypeError('name must be an instance of %s' % (string_type.__name__,))
```

Comment:
I'd keep it closer to pymongo's error message (e.g: "InvalidName: database names cannot contain the character '.'")
I'd suggest to use `TypeError(database name must be an instance of %s' % (string_type.__name__,))`, what do you think?

Reply:
Pymongo would throw TypeError so I duplicated that (no ways to directly reuse that piece of code). It's in database.py, line 106.